.cloudbuild/prombench.yaml

```diff
@@ -0,0 +1,335 @@
+
+  - id: "Setting environment variables"
+    name: gcr.io/cloud-builders/gcloud
+    env:
+    - '_PR_COMMENT_TEXT=$_PR_COMMENT_TEXT'
+    entrypoint: "bash"
+    args:
+      - "-c"
+      - |
+        COLLECTOR_TAG_OR_CANCEL=$(echo "${_PR_COMMENT_TEXT}" | head -n 1 | awk '{ print $2 }')
+        if [ "$${COLLECTOR_TAG_OR_CANCEL}" == "cancel" ]
+        then
+          
+          echo "Cancelling the benchmark run for PR #${_PR_NUMBER}."
+          echo "true" > /workspace/CANCEL_BENCHMARK
+
+        else
+          
+          echo "GMP Collector Tag: $${COLLECTOR_TAG_OR_CANCEL}"
+
+          if [[ "$${COLLECTOR_TAG_OR_CANCEL}" == '' ]]
+          then
+            echo "Since the collector version is missing, the latest one would be used."
+            DEFAULT_COLLECTOR_TAG=$(gcloud container images list-tags "$${PROMETHEUS_IMAGE_REPOSITORY}" --sort-by=UPDATE_TIME,TAGS | tail -1 | awk '{ print $2 }')
+            echo "$${DEFAULT_COLLECTOR_TAG}" > /workspace/PROMETHEUS_IMAGE_VERSION
+            export PROMETHEUS_IMAGE_VERSION="$${DEFAULT_COLLECTOR_TAG}"
+          else
+            GCLOUD_TAG=$(gcloud container images list-tags --filter="tags:$${COLLECTOR_TAG_OR_CANCEL}" --format=json $${PROMETHEUS_IMAGE_REPOSITORY})
+            if [[ "$${GCLOUD_TAG}" == "[]" ]]
+            then
+              echo "The $${PROMETHEUS_IMAGE_REPOSITORY} image does not have the $${COLLECTOR_TAG_OR_CANCEL} tag. Hence, this run will use the default latest tag."
+              DEFAULT_COLLECTOR_TAG=$(gcloud container images list-tags "$${PROMETHEUS_IMAGE_REPOSITORY}" --sort-by=UPDATE_TIME,TAGS | tail -1 | awk '{ print $2 }')
+              echo "$${DEFAULT_COLLECTOR_TAG}" > /workspace/PROMETHEUS_IMAGE_VERSION
+              export PROMETHEUS_IMAGE_VERSION="$${DEFAULT_COLLECTOR_TAG}"
+            else
+              echo "$${COLLECTOR_TAG_OR_CANCEL}" > /workspace/PROMETHEUS_IMAGE_VERSION
+              export PROMETHEUS_IMAGE_VERSION="$${COLLECTOR_TAG_OR_CANCEL}"
+            fi
+          fi
+          
+          echo "The release/base GMP Collector's tag is set to: $${PROMETHEUS_IMAGE_VERSION}."
+
+        fi
+
+  - id: "Checking cluster's existance"
+    name: gcr.io/cloud-builders/gcloud
+    entrypoint: "bash"
+    args:
+      - "-c"
+      - |
+        if $(gcloud container clusters describe "$${CLUSTER_NAME}" --project="$${GKE_PROJECT_ID}" --location="$${ZONE}" >/dev/null 2>&1)
+        then
+          CREATE_CLUSTER="false"
+          echo "$${CREATE_CLUSTER}" > /workspace/CREATE_CLUSTER
+          echo "The cluster (name: $${CLUSTER_NAME}) already exists. Therefore, skipping cluster creation."
+        else
+          CREATE_CLUSTER="true"
+          echo "$${CREATE_CLUSTER}" > /workspace/CREATE_CLUSTER
+          echo "The cluster (name: $${CLUSTER_NAME}) does not exist, the cluster will get created before creating the nodes and triggering the benchmark process."
+        fi
+
+  - id: "Setting gcloud credentials for triggering the benchmark"
```

Comment:
The previous gcloud commands get the authentication credentials from GCE metadata server. However, we need to have credentials stored locally for prombench (GCP APIs called in prombench after this step). The reason that this is not before the first gcloud command is that we only need this file if all the previous steps succeed. Otherwise, we need not store the credentials locally if any step fails (which fails the entire build).